ReactSkia/sdk/OnScreenKeyBoard.h

```diff
@@ -0,0 +1,175 @@
+    TOGGLE_KEY ,
+    TEXT_KEY,
+    FUNCTION_KEY,
+    KEY_TYPE_COUNT
+};
+
+//Configuration to be specified while OSK Launch
+struct OSKConfig {
+    OSKTypes        oskType;
+    OSKThemes       oskTheme;
+    const char *    placeHolderName;
+    const char *    returnKeyLabel;
+    bool      enablesReturnKeyAutomatically;
+};
+
+// Default OSK Configuration
+static OSKConfig defaultOSKConfig={ OSK_DEFAULT_TYPE, OSK_DEFAULT_THEME, nullptr, "done", false };
+
+typedef struct keyPosition {
+    SkPoint        textXY{}; // text X,Y to draw
+    SkPoint        textHLXY{}; // Text X,Y to draw on Highlight Tile
+    SkPoint        textCapsHLXY{}; // Text X,Y for Upper Case Alphabets
+    SkRect         highlightTile; // Highlight Tile coverage
+}keyPosition_t;
+
+typedef struct KeyInfo {
+    const char *   keyName;
+    rnsKey         keyValue;
+    keyType        keyType;
+    unsigned int   KBPartitionId;
+}KeyInfo_t;
+
+typedef struct KeySiblingInfo {
+    SkPoint siblingRight;
+    SkPoint siblingLeft;
+    SkPoint siblingUp;
+    SkPoint siblingDown;
+}keySiblingInfo_t;
+
+typedef struct keyPlacementConfig {
+    SkPoint groupOffset;
+    SkPoint groupKeySpacing;
+    float   hlTileFontSizeMultiplier;
+    float   fontScaleFactor;
+    unsigned int maxTextLength;
+}keyPlacementConfig_t;
+
+typedef std::vector<std::vector<KeyInfo_t>> KBLayoutKeyInfoContainer;
+typedef std::vector<std::vector<keyPosition_t>> KBLayoutKeyPosContainer;
+typedef std::vector<std::vector<keySiblingInfo_t>> KBLayoutSibblingInfoContainer;
+
+
+struct OSKLayout {
+    KBLayoutKeyInfoContainer*  keyInfo;
+    KBLayoutKeyPosContainer*    keyPos;
+    KBLayoutSibblingInfoContainer*    siblingInfo;
+    keyPlacementConfig_t*          KBGroupConfig;
+    KBLayoutType      KBLayoutType;
+    unsigned int      textFontSize;
+    unsigned int      textHLFontSize;
+    unsigned int      HorizontalStartOffset;
+    SkPoint           defaultFocussIndex;
+};
+
+class OnScreenKeyboard {
+
+    public:
+        static OnScreenKeyboard& getInstance(); // Interface to get OSK singleton object
+        static OSKErrorCode launch(OSKConfig oskConfig=defaultOSKConfig);// Interface to launch OSK
+        static void exit(); //Interface to quit OSK
+        static bool IsKBActive() {
+             OnScreenKeyboard &oskHandle=OnScreenKeyboard::getInstance();
+             return oskHandle.isOSKActive;
+        } // Interface to get whether is in launch or exit state
+
+    private:
+        OnScreenKeyboard() {
+            sem_init(&semPermitToLaunch,0,1);//Allow by default. Next signal to be done on exit call
+            sem_init(&semPermitToExit,0,0);//Block by default. signal to be done on launch Call
+        };
+        ~OnScreenKeyboard() {
+            sem_destroy(&semPermitToLaunch);
+            sem_destroy(&semPermitToExit);
+        };
+
+        void cleanUpOSKInstance();
+        void prepareToLaunch();
+        void createOSkWindow();
+        void showOSKWindow();
```

Comment:
Just a thought. Probably can be done as part of improvement as well.
Need to see possibility of having to create, show OSK Window as a utility APIs . This would help when we are adding more modules requiring separate window like Alert.

Reply:
Valid Suggestion. Will make note of it for future enhancement.